crates/services/consensus_module/poa/src/service_test/trigger_tests.rs

```diff
@@ -299,3 +333,144 @@ async fn interval_trigger_doesnt_react_to_full_txpool() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn interval_trigger_produces_blocks_in_the_future_when_time_is_lagging() {
+    // Given
```

Comment:
nit: something missing here?

Reply:
Not sure what you refer to here. The test is broken down into three sections "Given, When, Then", and this is the beginning of the first one.